pylint/testutils.py

```diff
@@ -622,6 +622,7 @@ def _split_lines(cls, expected_messages, lines):
 
     def _check_output_text(self, expected_messages, expected_lines, received_lines):
         expected_lines = self._split_lines(expected_messages, expected_lines)[0]
-        assert (
-            expected_lines == received_lines
-        ), "Expected test lines did not match for test: {}".format(self._test_file.base)
+        for expected, received in zip(expected_lines, received_lines):
+            assert (
+                expected == received
+            ), f"In '{self._test_file.base}' first unmatched lines: {expected} != {received}"
```

Comment:
Glad that this error message is getting enhanced, I've been putting breakpoints in and inspecting the vars when this check fails since the statement printed is quite unhelpful. It might be good to use sets here to show all the differences at once: 
```suggestion
            assert (
                expected_lines == received_lines
            ), f"'{self._test_file.base}' has missing lines: {set(expected_lines) - set(received_lines)},  and unexpected lines: {set(received_lines) - set(expected_lines)}"
```

Reply:
Definitely, thank you for the suggestion

Reply:
Actually I think your suggestion also revealed a bug in the change I made. I was not catching unexpected messages at the end of the files... So yeah, great catch.